### ISSUE TITLE
Game crashed as I died, and no grave was spawned in

### ISSUE THREAD

**Issue by That1RagingBat** [B1n-ry/Youre-in-grave-danger#183]
Hello, it’s as the title says. I died while fighting the masked chieftain boss(forgot his name, but it’s the Mowzie’s Mobs one), and when I died, my game hard crashed, and a grave stone was never made. I checked the list of graves and deaths, and it didn’t even show up, same with several others actually. The experience orb spawned, but no grave site like I said. I’m only making a fuss about it since I died with about a day’s work of stuff on me, mostly dragon bits. I’m not sure how it could be fixed, but if I could somehow restore it in game, that’d be great. Sorry to bother you

**Comment by That1RagingBat** [comment]
Upon recreating it, with the crash on death, it doesn’t spawn in a grave site at all. I could check if there’s somehow a cave under where I died for the test, but I dunno anymore at this point

**Comment by B1n-ry** [comment]
I'm afraid there might not be much I can do about getting your stuff back. It'd be great if you could share the crash log however. That way I can identify what's wrong, and fix it in future releases.
Also it would help to know what mod version and mod loader you are using

**Comment by BlazingTwist** [comment]
Can confirm that this occurs on Minecraft 1.21.1 when using the Accessories mod.
It seems the issue is now fixed in Accessories 1.1.0-beta.15
see https://github.com/wisp-forest/accessories/commit/c659649b0950fd77382383b4d573ce36d8f3b1a4

### Steps to reproduce
In case you want to troubleshoot this as well, since the issue only occurs when both Accessories and YIGD are installed.

Minimum required mods:
- [Accessories 1.1.0-beta.14](https://modrinth.com/mod/accessories/version/5CkHnpcM)
- [yigd 2.4.11](https://modrinth.com/mod/yigd/version/2.4.11)
- [owo-lib 0.12.15](https://modrinth.com/mod/owo-lib/version/0.12.15+1.21)
- [fabric-api 0.106.0](https://modrinth.com/mod/fabric-api/version/0.106.0+1.21.1)
- [cloth-config 15.0.140](https://modrinth.com/mod/cloth-config/version/15.0.140+fabric)

Get killed by a skeleton (dying from e.g. Fall Damage does not trigger the exception)
The exception occurs even if your Inventory is empty.

<details>

<summary>Exception log</summary>

```
---- Minecraft Crash Report ----
// I bet Cylons wouldn't have this problem.

Time: 2024-10-26 13:01:54
Description: Ticking entity

java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "c" is null
	at java.base/java.util.ArrayList.addAll(ArrayList.java:752)
	at knot//io.wispforest.accessories.impl.caching.AccessoriesHolderLookupCache.getAllEquipped(AccessoriesHolderLookupCache.java:100)
	at knot//io.wispforest.accessories.impl.AccessoriesCapabilityImpl.getAllEquipped(AccessoriesCapabilityImpl.java:353)
	at knot//io.wispforest.accessories.api.AccessoriesCapability.getAllEquipped(AccessoriesCapability.java:246)
	at knot//net.minecraft.class_1890.handler$zzk000$accessories$adjustIterationWithAccessories(class_1890.java:702)
	at knot//net.minecraft.class_1890.method_8209(class_1890.java:155)
	at knot//net.minecraft.class_1890.method_60619(class_1890.java:203)
	at knot//net.minecraft.class_1665.method_7454(class_1665.java:415)
	at knot//net.minecraft.class_1676.method_7488(class_1676.java:208)
	at knot//net.minecraft.class_1676.method_59860(class_1676.java:182)
	at knot//net.minecraft.class_1665.method_5773(class_1665.java:248)
	at knot//net.minecraft.class_1667.method_5773(class_1667.java:74)
	at knot//net.minecraft.class_3218.method_18762(class_3218.java:770)
	at knot//net.minecraft.class_1937.method_18472(class_1937.java:498)
	at knot//net.minecraft.class_3218.method_31420(class_3218.java:408)
	at knot//net.minecraft.class_5574.method_31791(class_5574.java:54)
	at knot//net.minecraft.class_3218.method_18765(class_3218.java:372)
	at knot//net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:1021)
	at knot//net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:912)
	at knot//net.minecraft.class_1132.method_3748(class_1132.java:114)
	at knot//net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:697)
	at knot//net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:281)
	at java.base/java.lang.Thread.run(Thread.java:1583)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at java.base/java.util.ArrayList.addAll(ArrayList.java:752)
	at knot//io.wispforest.accessories.impl.caching.AccessoriesHolderLookupCache.getAllEquipped(AccessoriesHolderLookupCache.java:100)
	at knot//io.wispforest.accessories.impl.AccessoriesCapabilityImpl.getAllEquipped(AccessoriesCapabilityImpl.java:353)
	at knot//io.wispforest.accessories.api.AccessoriesCapability.getAllEquipped(AccessoriesCapability.java:246)
	at knot//net.minecraft.class_1890.handler$zzk000$accessories$adjustIterationWithAccessories(class_1890.java:702)
	at knot//net.minecraft.class_1890.method_8209(class_1890.java:155)
	at knot//net.minecraft.class_1890.method_60619(class_1890.java:203)
	at knot//net.minecraft.class_1665.method_7454(class_1665.java:415)
	at knot//net.minecraft.class_1676.method_7488(class_1676.java:208)
	at knot//net.minecraft.class_1676.method_59860(class_1676.java:182)
	at knot//net.minecraft.class_1665.method_5773(class_1665.java:248)
	at knot//net.minecraft.class_1667.method_5773(class_1667.java:74)
	at knot//net.minecraft.class_3218.method_18762(class_3218.java:770)
	at knot//net.minecraft.class_1937.method_18472(class_1937.java:498)
	at knot//net.minecraft.class_3218.method_31420(class_3218.java:408)
	at knot//net.minecraft.class_5574.method_31791(class_5574.java:54)

-- Entity being ticked --
Details:
	Entity Type: minecraft:arrow (net.minecraft.class_1667)
	Entity ID: 644
	Entity Name: Arrow
	Entity's Exact location: -0.82, 67.60, 5.99
	Entity's Block location: World: (-1,67,5), Section: (at 15,3,5 in -1,4,0; chunk contains blocks -16,-64,0 to -1,319,15), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
	Entity's Momentum: 1.27, -0.11, 0.86
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at knot//net.minecraft.class_1937.method_18472(class_1937.java:498)
	at knot//net.minecraft.class_3218.method_31420(class_3218.java:408)
	at knot//net.minecraft.class_5574.method_31791(class_5574.java:54)
	at knot//net.minecraft.class_3218.method_18765(class_3218.java:372)
	at knot//net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:1021)
	at knot//net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:912)
	at knot//net.minecraft.class_1132.method_3748(class_1132.java:114)
	at knot//net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:697)
	at knot//net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:281)
	at java.base/java.lang.Thread.run(Thread.java:1583)

-- Affected level --
Details:
	All players: 1 total; [class_3222['BlazingTwist'/54, l='ServerLevel[Accessories]', x=0.02, y=66.00, z=6.78]]
	Chunk stats: 2601
	Level dimension: minecraft:overworld
	Level spawn location: World: (0,75,0), Section: (at 0,11,0 in 0,4,0; chunk contains blocks 0,-64,0 to 15,319,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 13175 game time, 13175 day time
	Level name: Accessories
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Commands: true
	Level weather: Rain time: 15040 (now: false), thunder time: 31769 (now: false)
	Known server brands: fabric
	Removed feature flags: 
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
	Loaded entity count: 272
Stacktrace:
	at knot//net.minecraft.class_3218.method_8538(class_3218.java:1693)
	at knot//net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:1024)
	at knot//net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:912)
	at knot//net.minecraft.class_1132.method_3748(class_1132.java:114)
	at knot//net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:697)
	at knot//net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:281)
	at java.base/java.lang.Thread.run(Thread.java:1583)

-- System Details --
Details:
	Minecraft Version: 1.21.1
	Minecraft Version ID: 1.21.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 21.0.2, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 146681504 bytes (139 MiB) / 1056964608 bytes (1008 MiB) up to 8589934592 bytes (8192 MiB)
	CPUs: 8
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
	Identifier: Intel64 Family 6 Model 60 Stepping 3
	Microarchitecture: Haswell (Client)
	Frequency (GHz): 4.00
	Number of physical packages: 1
	Number of physical CPUs: 4
	Number of logical CPUs: 8
	Graphics card #0 name: Intel(R) HD Graphics 4600
	Graphics card #0 vendor: Intel Corporation
	Graphics card #0 VRAM (MiB): 1024.00
	Graphics card #0 deviceId: VideoController1
	Graphics card #0 versionInfo: 20.19.15.5171
	Graphics card #1 name: NVIDIA GeForce RTX 3060
	Graphics card #1 vendor: NVIDIA
	Graphics card #1 VRAM (MiB): 12288.00
	Graphics card #1 deviceId: VideoController2
	Graphics card #1 versionInfo: 31.0.15.3734
	Memory slot #0 capacity (MiB): 8192.00
	Memory slot #0 clockSpeed (GHz): 1.80
	Memory slot #0 type: DDR3
	Memory slot #1 capacity (MiB): 8192.00
	Memory slot #1 clockSpeed (GHz): 1.80
	Memory slot #1 type: DDR3
	Memory slot #2 capacity (MiB): 8192.00
	Memory slot #2 clockSpeed (GHz): 1.80
	Memory slot #2 type: DDR3
	Memory slot #3 capacity (MiB): 8192.00
	Memory slot #3 clockSpeed (GHz): 1.80
	Memory slot #3 type: DDR3
	Virtual memory max (MiB): 47365.36
	Virtual memory used (MiB): 35093.75
	Swap memory total (MiB): 14722.49
	Swap memory used (MiB): 2640.31
	Space in storage for jna.tmpdir (MiB): <path not set>
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): <path not set>
	Space in storage for io.netty.native.workdir (MiB): <path not set>
	Space in storage for java.io.tmpdir (MiB): available: 333059.31, total: 953867.00
	Space in storage for workdir (MiB): available: 333059.31, total: 953867.00
	JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms512m -Xmx8192m
	Fabric Mods: 
		accessories: Accessories 1.1.0-beta.14+1.21.1
		cloth-config: Cloth Config v15 15.0.140
			cloth-basic-math: cloth-basic-math 0.6.1
		fabric-api: Fabric API 0.106.0+1.21.1
			fabric-api-base: Fabric API Base 0.4.42+6573ed8c19
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.68+b559734419
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.29+5bd9f1bc19
			fabric-block-api-v1: Fabric Block API (v1) 1.0.22+0af3f5a719
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.10+6573ed8c19
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.52+0af3f5a719
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.15+6573ed8c19
			fabric-command-api-v1: Fabric Command API (v1) 1.2.49+f71b366f19
			fabric-command-api-v2: Fabric Command API (v2) 2.2.28+6ced4dd919
			fabric-commands-v0: Fabric Commands (v0) 0.2.66+df3654b319
			fabric-content-registries-v0: Fabric Content Registries (v0) 8.0.16+b559734419
			fabric-convention-tags-v1: Fabric Convention Tags 2.1.0+7f945d5b19
			fabric-convention-tags-v2: Fabric Convention Tags (v2) 2.9.0+83a9d31319
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.29+0af3f5a719
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.1.28+1daea21519
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 20.2.21+16c4ae2519
			fabric-dimensions-v1: Fabric Dimensions API (v1) 4.0.0+6fc22b9919
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.7.0+2122d82819
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.7.13+ba9dae0619
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.53+6ced4dd919
			fabric-item-api-v1: Fabric Item API (v1) 11.1.0+d5debaed19
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.1.5+3fc0e55219
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.47+0af3f5a719
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.45+df3654b319
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.3.12+6c1df36019
			fabric-loot-api-v2: Fabric Loot API (v2) 3.0.14+3f89f5a519
			fabric-loot-api-v3: Fabric Loot API (v3) 1.0.2+3f89f5a519
			fabric-message-api-v1: Fabric Message API (v1) 6.0.13+6573ed8c19
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 2.0.0+fe474d6b19
			fabric-networking-api-v1: Fabric Networking API (v1) 4.3.0+c7469b2119
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 15.2.0+40875a9319
			fabric-particles-v1: Fabric Particles (v1) 4.0.2+6573ed8c19
			fabric-recipe-api-v1: Fabric Recipe API (v1) 5.0.13+6508971219
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 5.1.3+60c3209b19
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.4.0+c705a49c19
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.7.0+c705a49c19
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.68+df3654b319
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.48+73761d2e19
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.1.6+1daea21519
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.71+df3654b319
			fabric-rendering-v1: Fabric Rendering (v1) 5.0.5+df16efd019
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 4.3.0+8dc279b119
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 1.3.0+5659912919
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.25+8b68f1c719
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.87+b559734419
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.23+6573ed8c19
			fabric-transfer-api-v1: Fabric Transfer API (v1) 5.4.0+95bead5f19
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 6.1.0+1daea21519
		fabricloader: Fabric Loader 0.16.7
			mixinextras: MixinExtras 0.4.1
		java: OpenJDK 64-Bit Server VM 21
		minecraft: Minecraft 1.21.1
		owo: oωo 0.12.15+1.21
			io_wispforest_endec: endec 0.1.5
			io_wispforest_endec_gson: gson 0.1.3
			io_wispforest_endec_jankson: jankson 0.1.3
			io_wispforest_endec_netty: netty 0.1.2
		yigd: You're in Grave Danger 2.4.11
			fabric-permissions-api-v0: fabric-permissions-api 0.3.1
			libgui: LibGui 11.0.0+1.21
				jankson: Jankson 8.0.0+j1.2.3
					blue_endless_jankson: jankson 1.2.3
				libninepatch: LibNinePatch 1.2.0
	Server Running: true
	Player Count: 1 / 8; [class_3222['BlazingTwist'/54, l='ServerLevel[Accessories]', x=0.02, y=66.00, z=6.78]]
	Active Data Packs: vanilla, fabric, accessories, fabric-convention-tags-v2, yigd
	Available Data Packs: accessories, bundle, fabric, fabric-convention-tags-v2, trade_rebalance, vanilla, yigd
	Enabled Feature Flags: minecraft:vanilla
	World Generation: Stable
	World Seed: -3529455507282441422
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fabric'; Server brand changed to 'fabric'
	Launched Version: 1.21.1
```
</details>

**Comment by B1n-ry** [comment]
Since it seems to be fixed I'll close the issue. Thank you for the clear issue description anyway! I really appreciate it!
